registry/quilt_server/views.py

```diff
@@ -335,6 +338,11 @@ def wrapper(*args, **kwargs):
                     assert user
                     email = data['email']
                     is_admin = data.get('is_staff', False)
+                    if require_admin and not is_admin:
```

Comment:
You should move this check further down, to just above `f(*args, **kwargs)`, and use `g.auth.is_admin`:
- Having the check inside of a `try` or an `else` clause makes it harder to prove that it will always run, and there isn't some code path that bypasses it.
- `require_admin` is more sensitive than `require_login`. Any endpoint that requires login probably relies on the user and will fail if there's no user. An admin endpoint, on the other hand, will likely succeed. So it's good to be very explicit.
- Relying on `require_login=True` is fragile - some day, the code might get refactored, etc.

(You could move the `if require_login` check, too, though, for consistency.)

Reply:
Good point, fixed